Emby.Server.Implementations/Syncplay/SyncplayController.cs

```diff
@@ -228,190 +271,271 @@ public void SessionJoin(SessionInfo session, JoinGroupRequest request)
+                var command = NewSyncplayCommand(SendCommandType.Play);
+                SendCommand(session, BroadcastType.CurrentSession, command, CancellationToken.None);
+            }
+        }
 
-                    var command = NewSyncplayCommand(SendCommandType.Pause);
-                    SendCommand(session, BroadcastType.AllGroup, command);
-                }
-                else
-                {
-                    var command = NewSyncplayCommand(SendCommandType.Pause);
-                    SendCommand(session, BroadcastType.SingleSession, command);
-                }
+        /// <summary>
+        /// Handles a pause action requested by a session.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        /// <param name="request">The pause action.</param>
+        private void HandlePauseRequest(SessionInfo session, PlaybackRequest request)
+        {
+            if (!_group.IsPaused)
+            {
+                // Pause group and compute the media playback position
+                _group.IsPaused = true;
+                var currentTime = DateTime.UtcNow;
+                var elapsedTime = currentTime - _group.LastActivity;
+                _group.LastActivity = currentTime;
+                // Seek only if playback actually started
+                // (a pause request may be issued during the delay added to account for latency)
+                _group.PositionTicks += elapsedTime.Ticks > 0 ? elapsedTime.Ticks : 0;
+
+                var command = NewSyncplayCommand(SendCommandType.Pause);
+                SendCommand(session, BroadcastType.AllGroup, command, CancellationToken.None);
             }
-            else if (request.Type.Equals(PlaybackRequestType.Seek))
+            else
             {
-                // Sanitize PositionTicks
-                var ticks = request.PositionTicks ??= 0;
-                ticks = ticks >= 0 ? ticks : 0;
-                if (_group.PlayingItem.RunTimeTicks != null)
-                {
-                    var runTimeTicks = _group.PlayingItem.RunTimeTicks ??= 0;
-                    ticks = ticks > runTimeTicks ? runTimeTicks : ticks;
-                }
+                // Client got lost, sending current state
+                var command = NewSyncplayCommand(SendCommandType.Pause);
+                SendCommand(session, BroadcastType.CurrentSession, command, CancellationToken.None);
+            }
+        }
 
+        /// <summary>
+        /// Handles a seek action requested by a session.
+        /// </summary>
+        /// <param name="session">The session.</param>
+        /// <param name="request">The seek action.</param>
+        private void HandleSeekRequest(SessionInfo session, PlaybackRequest request)
+        {
+            // Sanitize PositionTicks
+            var ticks = SanitizePositionTicks(request.PositionTicks);
+
+            // Pause and seek
+            _group.IsPaused = true;
```

Comment:
why pause?..

Reply:
Mainly because of network bandwidth, which in turn affects when clients are ready. Seeking and expecting players to be playing the media that fast results in the content being skipped. The time clients spend on buffering is counted by the server as "playing time".

This will probably change when clients implement "group waiting": that functionality hasn't been tested yet so I don't really know how it behaves.

So, should this change be made now, as to not break the "specs" later, or is it fine changing it later?

Reply:
We can change it later. Spec will only get finalized in a sense only when a release is made. We can always do a few more PRs until then :)